tests/MongoDB.Analyzer.Tests.Common.TestCases/Builders/BuildersDefinitions.cs

```diff
@@ -0,0 +1,188 @@
+                (Builders<User>.Filter.Ne(u => u.Name, "Bob") &
+                 Builders<User>.Filter.Exists(u => u.LastName)));
+
+            _ = Builders<User>.Filter.Lt(u => u.Age, 10) |
+                Builders<User>.Filter.Gt(u => u.Age, 20) |
+                (filterDefinition.Ne(u => u.Name, "Bob") &
+                 filterDefinition.Exists(u => u.LastName));
+        }
+
+        [BuildersMQL("{ \"Address\" : -1 }")]
+        [BuildersMQL("{ new { Field = \"Name\", IntValue = 10, BoolValue = false, DoubleValue = 2.5 }.Field : \"2d\", new { Field = \"LastName\", IntValue = 10, BoolValue = false, DoubleValue = 2.5 }.Field : \"2d\", new { Field = \"Address\", IntValue = 10, BoolValue = false, DoubleValue = 2.5 }.Field : \"text\", new { Field = \"Vehicle\", IntValue = 10, BoolValue = false, DoubleValue = 2.5 }.Field : \"text\" }")]
+        [BuildersMQL("{ \"Name\" : \"2d\", \"LastName\" : \"2d\", \"Address\" : \"text\", \"Vehicle\" : \"text\" }")]
+        [BuildersMQL("{ new { Field = \"Name\", IntValue = 10, BoolValue = false, DoubleValue = 2.5 }.Field : \"2d\", new { Field = \"LastName\", IntValue = 10, BoolValue = false, DoubleValue = 2.5 }.Field : \"2d\", new { Field = \"Address\", IntValue = 10, BoolValue = false, DoubleValue = 2.5 }.Field : \"text\", new { Field = \"Vehicle\", IntValue = 10, BoolValue = false, DoubleValue = 2.5 }.Field : \"text\" }")]
+        public void IndexKeysDefinition()
+        {
+            var indexDefinition = Builders<Person>.IndexKeys;
+
+            _ = indexDefinition.Descending(x => x.Address);
+
+            _ = indexDefinition.Combine(
+                    indexDefinition.Geo2D(new { Field = "Name", IntValue = 10, BoolValue = false, DoubleValue = 2.5 }.Field)
+                        .Geo2D(new { Field = "LastName", IntValue = 10, BoolValue = false, DoubleValue = 2.5 }.Field),
+                    indexDefinition.Text(new { Field = "Address", IntValue = 10, BoolValue = false, DoubleValue = 2.5 }.Field)
+                        .Text(new { Field = "Vehicle", IntValue = 10, BoolValue = false, DoubleValue = 2.5 }.Field));
+
+            _ = indexDefinition.Combine(Builders<Person>.IndexKeys.Geo2D(u => u.Name).Geo2D("LastName"), Builders<Person>.IndexKeys.Text(u => u.Address).Text("Vehicle"));
+
+            _ = Builders<Person>.IndexKeys.Combine(
+                    indexDefinition.Geo2D(new { Field = "Name", IntValue = 10, BoolValue = false, DoubleValue = 2.5 }.Field)
+                        .Geo2D(new { Field = "LastName", IntValue = 10, BoolValue = false, DoubleValue = 2.5 }.Field),
+                    indexDefinition.Text(new { Field = "Address", IntValue = 10, BoolValue = false, DoubleValue = 2.5 }.Field)
+                        .Text(new { Field = "Vehicle", IntValue = 10, BoolValue = false, DoubleValue = 2.5 }.Field));
+        }
+
+        [BuildersMQL("{ \"Name\" : \"User Name1\" }")]
+        [BuildersMQL("{ \"Name\" : \"User Name2\" }")]
+        public void MethodTest()
```

Comment:
No need for 'Test' suffix. Let's try come up with a better convention here. Something like

Variable_definition_projection,
Variable_definition_indexkeys,
Method_definition
Static_method_definition
...

Reply:
Done